src/otaclient_iot_logging_server/log_proxy_server.py

```diff
@@ -17,79 +17,89 @@
 
 import asyncio
 import logging
-import time
-from http import HTTPStatus
-from queue import Full, Queue
+from concurrent.futures import ThreadPoolExecutor
 
+import grpc.aio
 from aiohttp import web
-from aiohttp.web import Request
 
-from otaclient_iot_logging_server._common import LogMessage, LogsQueue
+from otaclient_iot_logging_server._common import LogsQueue
 from otaclient_iot_logging_server._sd_notify import (
     READY_MSG,
     sd_notify,
     sd_notify_enabled,
 )
 from otaclient_iot_logging_server.configs import server_cfg
 from otaclient_iot_logging_server.ecu_info import ecu_info
+from otaclient_iot_logging_server.servicer import OTAClientIoTLoggingServerServicer
+from otaclient_iot_logging_server.v1 import (
+    otaclient_iot_logging_server_v1_pb2_grpc as v1_grpc,
+)
+from otaclient_iot_logging_server.v1.api_stub import OTAClientIoTLoggingServiceV1
 
 logger = logging.getLogger(__name__)
 
 
-class LoggingPostHandler:
-    """A simple aiohttp server handler that receives logs from otaclient."""
-
-    def __init__(self, queue: LogsQueue) -> None:
-        self._queue = queue
-        self._allowed_ecus = None
-
-        if ecu_info:
-            self._allowed_ecus = ecu_info.ecu_id_set
-            logger.info(
-                f"setup allowed_ecu_id from ecu_info.yaml: {ecu_info.ecu_id_set}"
-            )
-        else:
-            logger.warning(
-                "no ecu_info.yaml presented, logging upload filtering is DISABLED"
-            )
-
-    # route: POST /{ecu_id}
-    async def logging_post_handler(self, request: Request):
-        """
-        NOTE: use <ecu_id> as log_stream_suffix, each ECU has its own
-              logging stream for uploading.
-        """
-        _ecu_id = request.match_info["ecu_id"]
-        _raw_logging = await request.text()
-        _allowed_ecus = self._allowed_ecus
-        # don't allow empty request or unknowned ECUs
-        # if ECU id is unknown(not listed in ecu_info.yaml), drop this log.
-        if not _raw_logging or (_allowed_ecus and _ecu_id not in _allowed_ecus):
-            return web.Response(status=HTTPStatus.BAD_REQUEST)
-
-        _logging_msg = LogMessage(
-            timestamp=int(time.time()) * 1000,  # milliseconds
-            message=_raw_logging,
-        )
-        # logger.debug(f"receive log from {_ecu_id}: {_logging_msg}")
-        try:
-            self._queue.put_nowait((_ecu_id, _logging_msg))
-        except Full:
-            logger.debug(f"message dropped: {_logging_msg}")
-            return web.Response(status=HTTPStatus.SERVICE_UNAVAILABLE)
-
-        return web.Response(status=HTTPStatus.OK)
-
-
 WAIT_BEFORE_SEND_READY_MSG = 2  # seconds
 
 
-def launch_server(queue: Queue[tuple[str, LogMessage]]) -> None:
-    handler = LoggingPostHandler(queue=queue)
+async def _start_http_server(queue: LogsQueue):
+    handler = OTAClientIoTLoggingServerServicer(ecu_info=ecu_info, queue=queue)
```

Comment:
I notice that the same `OTAClientIoTLoggingServerServicer` instance has been created for both `start_http_server` and `start_grpc_server`, and actually each instance contains the same queue and same ecu_info.

What about changing as follow:
1. in start_server below, 
```python
async def _start_server(queue: LogsQueue):
    handler = OTAClientIoTLoggingServerServicer(ecu_info=ecu_info, queue=queue)
    await asyncio.gather(
        _start_http_server(handler),
        _start_grpc_server(handler),
    )
```
2. for `start_http_server` and `start_grpc_server`, directly take handler as param.